src/py/flwr/client/supernode/app.py

```diff
@@ -40,12 +40,14 @@
 from flwr.common.exit_handlers import register_exit_handlers
 from flwr.common.logger import log, warn_deprecated_feature
 
-from ..app import start_client_internal
 from ..clientapp.utils import get_load_client_app_fn
 
 
 def run_supernode() -> None:
     """Run Flower SuperNode."""
+    # Import start_client_internal here to avoid circular imports
+    from ..app import start_client_internal  # pylint: disable=import-outside-toplevel
```

Comment:
What's the circular import here? I'd like to understand it better. Perhaps we can find a different solution.